pkg/apiserver/api_installer.go

```diff
@@ -216,18 +216,27 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 	}
 	versionedStatus := indirectArbitraryPointer(versionedStatusPtr)
 	var (
-		getOptions          runtime.Object
-		versionedGetOptions runtime.Object
-		getOptionsKind      string
-		getSubpath          bool
-		getSubpathKey       string
+		getOptions             runtime.Object
+		versionedGetOptions    runtime.Object
+		getOptionsInternalKind unversioned.GroupVersionKind
+		getOptionsExternalKind unversioned.GroupVersionKind
+		getSubpath             bool
+		getSubpathKey          string
 	)
 	if isGetterWithOptions {
 		getOptions, getSubpath, getSubpathKey = getterWithOptions.NewGetOptions()
-		_, getOptionsKind, err = a.group.Typer.ObjectVersionAndKind(getOptions)
+		getOptionsGVString, getOptionsKind, err := a.group.Typer.ObjectVersionAndKind(getOptions)
```

Comment:
> I prefer the old way, using the reqScope.InternalVersion, which must be correct.

I think you may have gotten the fields backwards.  The reason we had to have the hack (different version) before was precisely because the `reqScope.InternalVersion` was never correct unless you were in the legacy kube group.  For every other group, your `reqScope.InternalVersion` is for your group, but your `getOptions` were for `/v1` (or `unversioned`).

This code is done in the order that always correct:  "Give me an instance of the `getOptions` type you expect and I'll determine the exact internal group."